influxql/engine.go

```diff
@@ -663,6 +665,31 @@ func ReduceMean(key Key, values []interface{}, e *Emitter) {
 	e.Emit(key, out.Sum/float64(out.Count))
 }
 
+// MapMin collects the values to pass to the reducer
+func MapMin(itr Iterator, e *Emitter, tmin int64) {
+	var values []float64
+
+	for k, v := itr.Next(); k != 0; k, v = itr.Next() {
+		values = append(values, v.(float64))
+	}
+	e.Emit(Key{tmin, itr.Tags()}, values)
+}
+
+// ReduceMin computes the min of value.
+func ReduceMin(key Key, values []interface{}, e *Emitter) {
+	var min float64
+	for i, value := range values {
+		vals := value.([]float64)
+		for j, v := range vals {
+			if i == 0 && j == 0 {
```

Comment:
Go has default values.  Min is `0` by default, so if all your values are above zero, it would return zero as the min falsely.  This is the easiest way I could come up with to "initialize" min to a real value, not the Go default value.